[ONNX] Fix sum export with attribute keepdims (#59316)

BowenBao opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #60249 [ONNX] Update documentation (#58712)
* #60248 [ONNX] shape type inference fixes for control flow (#59319)
* #60247 [ONNX] Extend chunk for dynamic chunk values (#59644)
* #60246 [ONNX] Add linspace symbolic (#58854)
* **#60245 [ONNX] Fix sum export with attribute keepdims (#59316)**
* #60244 [ONNX] Fix shape inference for large model (#59320)
* #60243 [ONNX] Handle onnx::Size in ComputeConstant folding (#59122)

Fix after b9bdb07a0261ab5a0b1038f290fa03af6ce0415f. Improving previous fix on two aspects
* Not only checks 0 on first dimension for empty tensor.
* Do not assume empty tensor when shape is not accessible.

Co-authored-by: BowenBao <bowbao@microsoft.com>

Differential Revision: [D29494917](https://our.internmc.facebook.com/intern/diff/D29494917)

BowenBao committed:
[ONNX] Fix sum export with attribute keepdims (#59316)

Fix after b9bdb07a0261ab5a0b1038f290fa03af6ce0415f. Improving previous fix on two aspects
* Not only checks 0 on first dimension for empty tensor.
* Do not assume empty tensor when shape is not accessible.

Co-authored-by: BowenBao <bowbao@microsoft.com>

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Fix sum export with attribute keepdims (#59316)"


Fix after b9bdb07a0261ab5a0b1038f290fa03af6ce0415f. Improving previous fix on two aspects
* Not only checks 0 on first dimension for empty tensor.
* Do not assume empty tensor when shape is not accessible.

Co-authored-by: BowenBao <bowbao@microsoft.com>

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Fix sum export with attribute keepdims (#59316)"


Fix after b9bdb07a0261ab5a0b1038f290fa03af6ce0415f. Improving previous fix on two aspects
* Not only checks 0 on first dimension for empty tensor.
* Do not assume empty tensor when shape is not accessible.

Co-authored-by: BowenBao <bowbao@microsoft.com>

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Fix sum export with attribute keepdims (#59316)"


Fix after b9bdb07a0261ab5a0b1038f290fa03af6ce0415f. Improving previous fix on two aspects
* Not only checks 0 on first dimension for empty tensor.
* Do not assume empty tensor when shape is not accessible.

Co-authored-by: BowenBao <bowbao@microsoft.com>

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Fix sum export with attribute keepdims (#59316)"


Fix after b9bdb07a0261ab5a0b1038f290fa03af6ce0415f. Improving previous fix on two aspects
* Not only checks 0 on first dimension for empty tensor.
* Do not assume empty tensor when shape is not accessible.

Co-authored-by: BowenBao <bowbao@microsoft.com>

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Fix sum export with attribute keepdims (#59316)"


Fix after b9bdb07a0261ab5a0b1038f290fa03af6ce0415f. Improving previous fix on two aspects
* Not only checks 0 on first dimension for empty tensor.
* Do not assume empty tensor when shape is not accessible.

Co-authored-by: BowenBao <bowbao@microsoft.com>

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Fix sum export with attribute keepdims (#59316)"


Fix after b9bdb07a0261ab5a0b1038f290fa03af6ce0415f. Improving previous fix on two aspects
* Not only checks 0 on first dimension for empty tensor.
* Do not assume empty tensor when shape is not accessible.

Co-authored-by: BowenBao <bowbao@microsoft.com>

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Fix sum export with attribute keepdims (#59316)"


Fix after b9bdb07a0261ab5a0b1038f290fa03af6ce0415f. Improving previous fix on two aspects
* Not only checks 0 on first dimension for empty tensor.
* Do not assume empty tensor when shape is not accessible.

Co-authored-by: BowenBao <bowbao@microsoft.com>

[ghstack-poisoned]

SplitInfinity commented:
@splitinfinity has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29494917).

BowenBao committed:
Update on "[ONNX] Fix sum export with attribute keepdims (#59316)"


Fix after b9bdb07a0261ab5a0b1038f290fa03af6ce0415f. Improving previous fix on two aspects
* Not only checks 0 on first dimension for empty tensor.
* Do not assume empty tensor when shape is not accessible.

Co-authored-by: BowenBao <bowbao@microsoft.com>

Differential Revision: [D29494917](https://our.internmc.facebook.com/intern/diff/D29494917)

[ghstack-poisoned]

SplitInfinity commented:
@splitinfinity has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29494917).
